storage/internal/apiv1/storage_client.go

```diff
@@ -0,0 +1,2388 @@
+	PatchBucket                      []gax.CallOption
+	UpdateBucket                     []gax.CallOption
+	StopChannel                      []gax.CallOption
+	DeleteDefaultObjectAccessControl []gax.CallOption
+	GetDefaultObjectAccessControl    []gax.CallOption
+	InsertDefaultObjectAccessControl []gax.CallOption
+	ListDefaultObjectAccessControls  []gax.CallOption
+	PatchDefaultObjectAccessControl  []gax.CallOption
+	UpdateDefaultObjectAccessControl []gax.CallOption
+	DeleteNotification               []gax.CallOption
+	GetNotification                  []gax.CallOption
+	InsertNotification               []gax.CallOption
+	ListNotifications                []gax.CallOption
+	DeleteObjectAccessControl        []gax.CallOption
+	GetObjectAccessControl           []gax.CallOption
+	InsertObjectAccessControl        []gax.CallOption
+	ListObjectAccessControls         []gax.CallOption
+	PatchObjectAccessControl         []gax.CallOption
+	UpdateObjectAccessControl        []gax.CallOption
+	ComposeObject                    []gax.CallOption
+	CopyObject                       []gax.CallOption
+	DeleteObject                     []gax.CallOption
+	GetObject                        []gax.CallOption
+	GetObjectMedia                   []gax.CallOption
+	InsertObject                     []gax.CallOption
+	ListObjects                      []gax.CallOption
+	RewriteObject                    []gax.CallOption
+	StartResumableWrite              []gax.CallOption
+	QueryWriteStatus                 []gax.CallOption
+	PatchObject                      []gax.CallOption
+	UpdateObject                     []gax.CallOption
+	GetObjectIamPolicy               []gax.CallOption
+	SetObjectIamPolicy               []gax.CallOption
+	TestObjectIamPermissions         []gax.CallOption
+	WatchAllObjects                  []gax.CallOption
+	GetServiceAccount                []gax.CallOption
+	CreateHmacKey                    []gax.CallOption
+	DeleteHmacKey                    []gax.CallOption
+	GetHmacKey                       []gax.CallOption
+	ListHmacKeys                     []gax.CallOption
+	UpdateHmacKey                    []gax.CallOption
+}
+
+func defaultGRPCClientOptions() []option.ClientOption {
+	return []option.ClientOption{
+		internaloption.WithDefaultEndpoint("storage.googleapis.com:443"),
+		internaloption.WithDefaultMTLSEndpoint("storage.mtls.googleapis.com:443"),
+		internaloption.WithDefaultAudience("https://storage.googleapis.com/"),
+		internaloption.WithDefaultScopes(DefaultAuthScopes()...),
+		internaloption.EnableJwtWithScope(),
+		option.WithGRPCDialOption(grpc.WithDisableServiceConfig()),
+		option.WithGRPCDialOption(grpc.WithDefaultCallOptions(
+			grpc.MaxCallRecvMsgSize(math.MaxInt32))),
+	}
+}
+
+func defaultCallOptions() *CallOptions {
+	return &CallOptions{
+		DeleteBucketAccessControl: []gax.CallOption{
+			gax.WithRetry(func() gax.Retryer {
+				return gax.OnCodes([]codes.Code{
+					codes.DeadlineExceeded,
+					codes.Unavailable,
```

Comment:
Are these retryers all just entirely defaults, or are they defined in service proto?

Reply:
They are defined in the grpc_service_config.json in [googleapis](https://github.com/googleapis/googleapis/blob/master/google/storage/v1/storage_grpc_service_config.json) by the AP team.

They are pretty basic. And retry doesn't apply to streaming methods _after_ the stream has been created (i.e. resumption).